source/common/common/logger.h

```diff
@@ -123,7 +124,72 @@ template <Id id> class Loggable {
 
 } // Logger
 
-#ifdef NDEBUG
+// Convert the line macro to a string literal for concatenation in log macros.
+#define DO_STRINGIZE(x) STRINGIZE(x)
+#define STRINGIZE(x) #x
+#define LINE_STRING DO_STRINGIZE(__LINE__)
+#define LOG_PREFIX __FILE__ ":" LINE_STRING "] "
+
+/**
+ * Base logging macros.  It is expected that users will use the convenience macros below rather than
+ * invoke these directly.
+ */
+#ifdef NVLOG
+#define log_trace_to_logger(LOGGER, ...)
+#define log_debug_to_logger(LOGGER, ...)
+#else
+#define log_trace_to_logger(LOGGER, ...) LOGGER.trace(LOG_PREFIX __VA_ARGS__)
+#define log_debug_to_logger(LOGGER, ...) LOGGER.debug(LOG_PREFIX __VA_ARGS__)
+#endif
+
+#define log_info_to_logger(LOGGER, ...) LOGGER.info(LOG_PREFIX __VA_ARGS__)
+#define log_warn_to_logger(LOGGER, ...) LOGGER.warn(LOG_PREFIX __VA_ARGS__)
+#define log_err_to_logger(LOGGER, ...) LOGGER.err(LOG_PREFIX __VA_ARGS__)
+#define log_critical_to_logger(LOGGER, ...) LOGGER.critical(LOG_PREFIX __VA_ARGS__)
+
+/**
+ * Convenience macro to log to a user-specified logger.
+ */
+#define log_to_logger(LOGGER, LEVEL, ...) log_##LEVEL##_to_logger(LOGGER, ##__VA_ARGS__)
+
+/**
+ * Convenience macro to log to the facility of the class in which the macro is invoked. Note:
+ * facilities are a way of grouping classes within envoy. A facility is specified by Id enum values
+ * listed above. The facility_log is enabled within a class when the class inherits from the
+ * Loggable class templated by the choice of facility.
+ */
+#define log_facility(LEVEL, ...) log_to_logger(log(), LEVEL, ##__VA_ARGS__)
```

Comment:
Still don't really like the term facility, but struggling with alternative suggestions:

* "service" <- too easy to confuse with other notions of service in Envoy.
* "category" <- make me think of category theory.
* "class/object" <- not necessarily coupled directly with class/object.
*  Just "log" <- already taken.

The other alternative is we switch to all caps, which we probably should with macros anyway, then we use `LOG` etc.

I'm out of ideas and I can probably live with `log_facility`.